src/components/Header.tsx

```diff
@@ -0,0 +1,170 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { useTodos } from '../hooks/useTodos';
+import * as postService from '../api/todos';
+import { USER_ID } from '../utils/USER_ID';
+
+export const Header: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const {
+    todos,
+    setTodos,
+    setErrorMessage,
+    setTempTodo,
+    todosInProcess,
+    setTodosInProcess,
+  } = useTodos();
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isInputDisabled, !todos?.length, todosInProcess]);
```

Comment:
Check this
![2023-10-10_11h35_05](https://github.com/mate-academy/react_todo-app-with-api/assets/61904995/194d7df9-2563-4415-80e5-c038e4c25661)